src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/NewVhdVMTests.ps1

```diff
@@ -0,0 +1,110 @@
+﻿# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+<#
+.SYNOPSIS
+Test New-AzureRmVhdVM with a valid disk file
+#>
+function Test-NewAzureRmVhdVMWithValidDiskFile
+{
+
+    # Setup
+    $rgname = Get-ComputeTestResourceName
+
+    try
+    {
+        # Common
+        [string]$loc = Get-ComputeVMLocation;
+        $loc = $loc.Replace(' ', '');
+
+        New-AzureRmResourceGroup -Name $rgname -Location $loc -Force;
+
+        # Create a new VM using the tiny VHD file
+        [string]$file = ".\VhdFiles\tiny.vhd";
+        $thrownException = $false;
+        $thrownErrorMessage = '';
+        try
+        {
+            $st = New-AzureRmVM -ResourceGroupName $rgname -Name $rgname -Location $loc -DiskFile $file;
```

Comment:
I don't understand - you are catching an exception when this is supposed to be a positive test case.  You are also not testing any properties of the VM.  Please include an actual positive test case in which you create a valid VM and validate the expected properties of the resulting object.

Reply:
I'm puzzled by the test framework as well, as you can see in the checkpoint in the next few lines, the test will fail with either 'queue empty' or 'object not found' errors only in playback mode.